src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -729,6 +729,19 @@ type KafkaAPITLS struct {
 // is the same as the name of the Cluster custom resource.
 type AdminAPITLS struct {
 	Enabled bool `json:"enabled,omitempty"`
+	// References cert-manager Issuer or ClusterIssuer. When provided, this
+	// issuer will be used to issue node certificates.
+	// Typically you want to provide the issuer when a generated self-signed one
+	// is not enough and you need to have a verifiable chain with a proper CA
+	// certificate.
+	IssuerRef *cmmeta.ObjectReference `json:"issuerRef,omitempty"`
+	// If provided, operator uses certificate in this secret instead of
+	// issuing its own node certificate. The secret is expected to provide
+	// the following keys: 'ca.crt', 'tls.key' and 'tls.crt'
+	// If NodeSecretRef points to secret in different namespace, operator will
```

Comment:
IMO it'd be better to either reject references to secrets in other namespaces or let the pod fail to schedule if an invalid reference is provided as that's more in the style of k8s.

This isn't end user facing though and I'm guessing that behavior is already in place because I see no diff related to it?

Reply:
Yeah, that was a behavior feature-flagged via an operator startup param, for cases where the secrets are shared with other resources in different namespaces.